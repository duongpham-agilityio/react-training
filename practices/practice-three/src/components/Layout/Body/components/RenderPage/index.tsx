import { Box, Spinner } from '@chakra-ui/react';
import { Suspense, memo } from 'react';
import { Outlet } from 'react-router-dom';

export const RenderPage = memo(() => (
  <Box flex={1}>
    <Suspense fallback={<Spinner />}>
      <Outlet />
    </Suspense>
  </Box>
));
