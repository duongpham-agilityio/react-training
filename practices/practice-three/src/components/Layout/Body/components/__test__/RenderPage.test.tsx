import { render } from '@testing-library/react';

// Components
import { RenderPage } from '../RenderPage';

describe('RenderPage', () => {
  it('Match to snapshot', () => {
    const { container } = render(<RenderPage />);

    expect(container).toMatchSnapshot();
  });
});
