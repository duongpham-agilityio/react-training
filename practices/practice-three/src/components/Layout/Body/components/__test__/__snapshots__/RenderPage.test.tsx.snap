// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderPage Match to snapshot 1`] = `
<div>
  <div
    class="css-1rr4qq7"
  />
</div>
`;
